File: BookstoreAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookstoreAPI.Models;

/// <summary>
/// Represents a book in the bookstore database.
/// </summary>
public class Book
{
    [Key]
    public int BookID { get; set; }

    [Required]
    public string Title { get; set; } = string.Empty;

    [Required]
    public string Author { get; set; } = string.Empty;

    [Required]
    public string Publisher { get; set; } = string.Empty;

    [Required]
    public string ISBN { get; set; } = string.Empty;

    [Required]
    public string Classification { get; set; } = string.Empty;

    [Required]
    public string Category { get; set; } = string.Empty;

    [Required]
    public int PageCount { get; set; }

    [Required]
    public decimal Price { get; set; }
}

File: client/src/types/Book.ts
/**
 * Book type matching the API response.
 */
export interface Book {
  bookID: number;
  title: string;
  author: string;
  publisher: string;
  isbn: string;
  classification: string;
  category: string;
  pageCount: number;
  price: number;
}

/**
 * Paginated books API response.
 */
export interface BooksResponse {
  books: Book[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}
